internal/context/context.go

```diff
@@ -198,3 +199,15 @@ func RootModuleWalker(ctx context.Context) (*rootmodule.Walker, error) {
 	}
 	return w, nil
 }
+
+func WithRootModuleLoader(rml rootmodule.RootModuleLoader, ctx context.Context) context.Context {
```

Comment:
Nit, make `ctx` first argument

Reply:
Yeah, I don't disagree, but I think I'd prefer to change all of the `With*` functions at the same time and perhaps do that in a separate PR, if that's ok and follow the same pattern throughout the package for now?